File: src/main/kotlin/com/example/externalDbOperations/DbLogin.java
package com.example.externalDbOperations;


import java.io.BufferedWriter;
import java.io.IOException;
import java.io.OutputStream;
import java.io.OutputStreamWriter;
import java.net.HttpURLConnection;
import java.net.URLEncoder;

public class DbLogin extends ExternalDbHelper
{
    private final String fileName = "login.php";

    private HttpURLConnection httpConnection;
    private String result;
    private String address;
    private String login;
    private String password;

    public DbLogin(String login, String password)
    {
        address = BASE_URL+fileName;
        this.login = login;
        this.password = password;
    }

    @Override
    public void run()
    {
        httpConnection = setConnection(address);
        sendData();
        result = readResult(httpConnection);
    }


    private void sendData()
    {
        try {
            OutputStream outStream = httpConnection.getOutputStream();
            BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(outStream, "UTF-8"));
            String dataToSend = URLEncoder.encode("user", "UTF-8") + "=" + URLEncoder.encode(login, "UTF-8")
                    + "&&" + URLEncoder.encode("pass", "UTF-8") + "=" + URLEncoder.encode(password, "UTF-8");

            writer.write(dataToSend);
            writer.flush();
            writer.close();
        }
        catch (IOException e)
        {
            System.out.println("sendData: "+e.getMessage());
        }
    }

    public String getResult()
    {
        try {
            join();
        } catch (InterruptedException e) {
            System.out.println("getResult: "+e.getMessage());
        }
        return result;
    }
}

File: src/main/kotlin/com/example/externalDbOperations/DbLogin.kt
package com.example.externalDbOperations

import java.io.BufferedWriter
import java.io.IOException
import java.io.OutputStreamWriter
import java.net.HttpURLConnection
import java.net.URLEncoder

class DbLogin(val login: String, val password: String) : ExternalDbHelper() {
    private val fileName = "login.php"
    private lateinit var httpConnection: HttpURLConnection
    private var result: String = ""
       // get() { this.join(); return field }
    private val address: String = BASE_URL + fileName


    override fun run() {
        sleep(2000)
        httpConnection = setConnection(address)
        sendData()
        result = readResult(httpConnection)
        onResultListener.onReceive(result)
    }

    private fun sendData() {
        try {
            val outStream = httpConnection.outputStream
            val writer = BufferedWriter(OutputStreamWriter(outStream, "UTF-8"))
            val dataToSend = (URLEncoder.encode("user", "UTF-8") + "=" + URLEncoder.encode(login, "UTF-8")
                    + "&&" + URLEncoder.encode("pass", "UTF-8") + "=" + URLEncoder.encode(password, "UTF-8"))
            writer.write(dataToSend)
            writer.flush()
            writer.close()
        } catch (e: IOException) {
            println("sendData: " + e.message)
        }
    }
}